mender.go

```diff
@@ -26,50 +26,78 @@ import (
 )
 
 type Controler interface {
-	GetState() MenderState
+	Bootstrap() error
+	TransitionState() MenderState
 	GetCurrentImageID() string
 	GetDaemonConfig() daemonConfig
 	GetUpdaterConfig() httpsClientConfig
+	LastError() error
 }
 
 const (
 	defaultManifestFile = "/etc/build_mender"
+	defaultKeyFile      = "/data/mender/mender-agent.pem"
 )
 
 type MenderState int
 
+// State transitions:
+//
+//            unknown
+//               |
+//               v
+//             init
+//               |
+//               v
+//          bootstrapped
+//               |
+//       +-------+-------------+
+//       |                     |
+//       v                     v
+// fresh update         wait for update
```

Comment:
Well, yes, but IMO it's a part of another task. As I've indicated here https://github.com/mendersoftware/mender/pull/58#discussion_r65037067 the client state model is incomplete and needs working.